### PR TITLE
Intrinsic method assembly should not be dropped on retry

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @kevinh-ms

**Comment by amcasey** [comment]
Attempted summary: On the first attempt, the intrinsic assembly is handed to us specially.  On subsequent assemblies, we have to dig it out of our references explicitly?

**Comment by cston** [comment]
@amcasey: On retry, we drop assemblies that are not directly referenced by the assembly containing the current frame. That meant we dropped the intrinsic method assembly as well since that is not explicitly referenced by any assembly.

**Comment by amcasey** [comment]
:+1:

File: src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs [hunk 1]
```diff
@@ -181,10 +190,11 @@ internal static bool DeclaresTheObjectClass(this MetadataReader reader)
         }
 
         /// <exception cref="BadImageFormatException">An exception from metadata reader.</exception>
-        private static bool IsSystemObjectOrThrow(this MetadataReader reader, TypeDefinition typeDef)
+        internal static bool IsPublicTypeOrThrow(this MetadataReader reader, TypeDefinition typeDef, string namespaceName, string typeName)
```
Comment (KevinH-MS) [inline src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs]: Should this just be "IsPublicType"?  I don't get the "OrThrow" bit (even in the old implementation).

Comment (cston) [inline src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs]: I believe "OrThrow" indicates to callers that the method may throw BadImageFormatException. @tmat

Comment (KevinH-MS) [inline src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs]: Actually, perhaps "IsPublicNonInterfaceType".

**Comment by KevinH-MS** [comment]
:+1:

File: src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs]: Renamed.

**Pull request merged**